src/app/components/core/control-dispatcher/api-flow/add-edit-api-flow/add-edit-api-flow.component.ts

```diff
@@ -0,0 +1,526 @@
+          (data: any) => {
+            /** request completed */
+            this.ngProgress.done();
+            this.alertService.success(data.message, true);
+            this.router.navigate(['control-dispatcher/api']);
+          },
+          error => {
+            /** request completed but error */
+            this.ngProgress.done();
+            if (error.status === 0) {
+              console.log('service down ', error);
+            } else {
+              this.alertService.error(error.statusText);
+            }
+          });
+    }
+  
+    requestAPIFlow(value) {
+        let variables = {};
+        value.variables.forEach(v => { variables[v.vName] = v.vValue });
+        this.controlAPIFlowService.requestAPIFlow(this.af.name, variables) 
+        .subscribe((data: any) => {
+            /** request completed */
+            this.ngProgress.done();
+            this.alertService.success(data.message, true);
+            this.closeModal('confirmation-execute-dialog');
+            this.getAPIFlow();
+            },
+            error => {
+            /** request completed but error */
+            this.ngProgress.done();
+            if (error.status === 0) {
+                console.log('service down ', error);
+            } else {
+                this.alertService.error(error.statusText);
+            }
+        });
+    }
+
+    public toggleDropdown(id) {
+        const activeDropDowns = Array.prototype.slice.call(document.querySelectorAll('.dropdown.is-active'));
+        if (activeDropDowns.length > 0) {
+          if (activeDropDowns[0].id !== id) {
+            activeDropDowns[0].classList.remove('is-active');
+          }
+        }
+        const dropDown = document.querySelector(`#${id}`);
+        dropDown.classList.toggle('is-active');
+    }
+
+    changeType(value) {
+        this.selectedType = value === 'Select Type' ? '' : value;
+    }
+
+    getDestTypes() {
+        this.controlAPIFlowService.getDestinationTypes()
+          .subscribe((data: any) => {
+            this.ngProgress.done();
+            // this.destinationTypes = data;
+            this.destinationTypes = data.filter(d => d.name !== 'Any');
+          }, error => {
+            if (error.status === 0) {
+              console.log('service down ', error);
+            } else {
+              this.alertService.error(error.statusText);
+            }
+          });
+    }
+
+    navigateToList() {
+        this.router.navigate(['control-dispatcher/api']);
+    }
+
+    onCancel() {
+        this.router.navigate(['control-dispatcher/api']);
+    }
+
+    onCheckboxClicked(event) {
+        this.af.anonymous = event.target.checked;
+    }
+
+    getFormControls(type): AbstractControl[] {
+        return (<FormArray>this.apiFlowForm.get(type)).controls;
+    }
+
+    selectDestinationName(value) {
+        this.selectedDestinationName = value === 'Select Destination Name' ? null : value;
+    }
+
+    getDestinationNames(selectedType) {
```

Comment:
this function cannot be used from a component ts? Many functions here are already in use in Control pipeline and scripts, users